xxxx-gpu-msm.md

```diff
@@ -0,0 +1,92 @@
+# RFC title
+
+GPU based MSM implementation.
+
+## Summary
+
+A rust library implementing gpu based acceleration for computing MSM, it can be integrated in kimchi and
+other projects that make use of MSMs with the pasta curves.
+
+## Motivation
+
+One of the mains components of kimchi's runtime is the multi scalar multiplication (MSM), it consist of
+basically taking a list of points, adding each point to itself N times, and then add all the resulting
+points together.
+This library will accelerate that process doing the computation in a gpu, it will also serve as a reference
+point to measure the improvements that gpu computations can provide us.
+
+This will build on top of an already existing implementation, and thus some of the design choices will be
+based on the fact that the particular decision is cheap to implement because it is already implemented.
+
+## Detailed design
```

Comment:
This section doesn't seem to mention the integration with kimchi in any detail. Can you add something about how that will look, what we require in terms of feature flags / feature detection /  etc.?

Reply:
added